presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/StreamPropertyDerivations.java

```diff
@@ -270,6 +270,12 @@ public StreamProperties visitTableScan(TableScanNode node, List<StreamProperties
             if (streamPartitionSymbols.isPresent() && streamPartitionSymbols.get().isEmpty()) {
                 return new StreamProperties(MULTIPLE, Optional.empty(), false);
             }
+            else if (streamPartitionSymbols.isPresent() && layout.getLocalProperties().stream().allMatch(property -> property.isOrderSensitive())) {
+                    Set<VariableReferenceExpression> orderedColumns = layout.getLocalProperties().stream().flatMap(property -> getNonConstantVariables(property.getColumns(), assignments, constants).get().stream()).collect(Collectors.toSet());
```

Comment:
We may need to only do this when the merge join is enabled.
Maybe add a comment:
This is to avoid adding an exchange between the join and scan when the merge join is enabled.